### PR TITLE
feat: Add mina-sftp integration tests & native support

### LINKED ISSUE #8333: Add mina-sftp integration tests & native support

### PR DESCRIPTION
**Pull request by gansheer**
Closes #8333 

* Create a common tests support module for sftp tests
* Native support for mina-sftp
* Add tests case from existing camel-ftp sftp use cases and some more complex ones
* Strengthen cipher, certificate, authentication and compression test
* Add script to recreate certificates used in the tests

Note: some of the dependencies and native specific code are due to camel-mina-sftp having camel-ftp dependency. Some of it should be revisted when we migrate to Camel >= 4.19.

### CODE REVIEW

File: extensions/mina-sftp/deployment/src/main/java/org/apache/camel/quarkus/component/mina/sftp/deployment/MinaSftpProcessor.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void filterFtpComponents(BuildProducer<CamelServiceFilterBuildItem> serviceFilter) {
+        // camel-mina-sftp depends on camel-ftp only for shared base classes (RemoteFileEndpoint, etc.)
+        serviceFilter.produce(new CamelServiceFilterBuildItem(CamelServiceFilter.forComponent("ftp")));
+        serviceFilter.produce(new CamelServiceFilterBuildItem(CamelServiceFilter.forComponent("ftps")));
+        serviceFilter.produce(new CamelServiceFilterBuildItem(CamelServiceFilter.forComponent("sftp")));
```
Comment (jamesnetherton) [inline extensions/mina-sftp/deployment/src/main/java/org/apache/camel/quarkus/component/mina/sftp/deployment/MinaSftpProcessor.java:40]: Not sure if we should add a note to docs that `camel-quarkus-ftp` wont work if added together with `camel-quarkus-mina-sftp`? It's a bit of an edge case. WDYT?

File: integration-tests/mina-sftp/src/test/java/org/apache/camel/quarkus/component/mina/sftp/it/MinaSftpTest.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    @Test
+    void loadComponentMinaSftp() {
+        /* A simple autogenerated test */
+        RestAssured.get("/mina-sftp/load/component/mina-sftp")
+                .then()
+                .statusCode(200);
+    }
+
+    @Test
+    public void testMinaSftpComponent() {
```
Comment (jamesnetherton) [inline integration-tests/mina-sftp/src/test/java/org/apache/camel/quarkus/component/mina/sftp/it/MinaSftpTest.java]: Nitpick - for any new test classes I try to not use `public` modifier.

File: integration-tests/mina-sftp/pom.xml [hunk 3]
```diff
@@ -53,9 +57,65 @@
...
+        <dependency>
+            <groupId>org.apache.sshd</groupId>
+            <artifactId>sshd-sftp</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.sshd</groupId>
+            <artifactId>sshd-scp</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (jamesnetherton) [inline integration-tests/mina-sftp/pom.xml]: Could probably be removed if they are transitives of `camel-quarkus-integration-tests-support-sftp`.

File: pom.xml [hunk 4]
```diff
@@ -88,6 +88,7 @@
         <commons-collections.version>${commons-collections-version}</commons-collections.version>
         <commons-collections4.version>${commons-collections4-version}</commons-collections4.version>
         <commons-exec.version>${commons-exec-version}</commons-exec.version>
+        <commons-net.version>3.12.0</commons-net.version><!-- @sync org.apache.camel:camel-ftp:${camel.version} dep:commons-net:commons-net -->
```
Comment (jamesnetherton) [inline pom.xml]: You can simplify and reference the property from the `camel-dependencies` parent:

```suggestion
        <commons-net.version>${commons-net-version}</commons-net.version>
```

Comment (gansheer) [inline pom.xml]: @ppalaga @jamesnetherton Is there a better way to use the version for the transitive dependence from camel ?

Comment (jamesnetherton) [reply pom.xml]: Yes - I mentioned it in my review comment. You can reuse the properties from `camel-dependencies`.

Comment (gansheer) [reply pom.xml]: True, my bad :sweat_smile:

File: extensions/mina-sftp/deployment/src/main/java/org/apache/camel/quarkus/component/mina/sftp/deployment/MinaSftpProcessor.java (hunk 1 above)
Comment (gansheer) [reply extensions/mina-sftp/deployment/src/main/java/org/apache/camel/quarkus/component/mina/sftp/deployment/MinaSftpProcessor.java:40]: Good idea, I added the limitation in the doc of both extensions

**Comment by jamesnetherton** [comment]
@gansheer please regenerate / sync the extension docs for `ftp` + `mina-sftp`.
